File: CMS/src/app/shared/AgencyMenu.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpParams } from '@angular/common/http';
import { environment } from 'src/environments/environment';
import { AgencyMenu } from 'src/app/shared/AgencyMenu.model';
import { BaseService } from './Base.service';
@Injectable({
  providedIn: 'root'
})
export class AgencyMenuService extends BaseService {
  List: AgencyMenu[] | undefined;
  ListFilter: AgencyMenu[] | undefined;
  FormData!: AgencyMenu;
  constructor(public httpClient: HttpClient) {
    super(httpClient);
    this.Controller = "AgencyMenu";
  }
}

File: Service/Implement/AgencyMenuService.cs
namespace Service.Implement
{
    public class AgencyMenuService : BaseService<AgencyMenu, IAgencyMenuRepository>
    , IAgencyMenuService
    {
    private readonly IAgencyMenuRepository _AgencyMenuRepository;
    public AgencyMenuService(IAgencyMenuRepository AgencyMenuRepository) : base(AgencyMenuRepository)
    {
    _AgencyMenuRepository = AgencyMenuRepository;
    }
    }
    }

